app/src/main/java/org/cru/godtools/ui/tools/ToolsAdapter.kt

```diff
@@ -61,7 +65,13 @@ class ToolsAdapter(lifecycleOwner: LifecycleOwner, viewModelProvider: ViewModelP
 
     override fun onCreateViewDataBinding(parent: ViewGroup, viewType: Int) = when (viewType) {
         VIEW_TYPE_LESSON -> ToolsListItemLessonBinding.inflate(LayoutInflater.from(parent.context), parent, false)
-        else -> ToolsListItemToolBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        else -> {
+            if(isAllTools){
+                DashboardListItemToolsBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+            }else {
+                ToolsListItemToolBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+            }
```

Comment:
you should add this as a new `viewType` in the `getItemViewType` method.

Reply:
the `viewType` affects how RecyclerView caches/reuses items. If in the future we make `isAllTools` dynamic, that would cause the `RecyclerView` cache to become corrupted and we would get a weird intermix of different layout types.